opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -13,25 +13,41 @@
  */
 package opentracing;
 
+
 /**
- * Tracer is a simple, thin interface for Span creation.
+ * Tracer is a simple, thin interface for Span creation, and Span propagation into different transport formats.
  */
 public interface Tracer {
 
   /**
-   * Create, start, and return a new Span with the given `operationName`, all without specifying a
-   * parent Span that can be used to incorporate the newly-returned Span into an existing trace.
+   * Create, start, and return a new Span with the given `operationName`.
+   * An optional parent Span can be specified used to incorporate the newly-returned Span into an existing trace.
    *
    * <p>Example:
    * <pre>{@code
    * Tracer tracer = ...
-   * Span feed = tracer.startTrace("GetFeed");
-   * Span http = tracer.startTrace("HandleHTTPRequest")
+   * Span feed = tracer.startTrace("GetFeed", null);
+   * Span http = tracer.startTrace("HandleHTTPRequest", feed)
    *                   .setTag("user_agent", req.UserAgent)
    *                   .setTag("lucky_number", 42);
    * }</pre>
    */
-  Span startTrace(String operationName);
+  Span startSpan(String operationName, /* @Nullable */ Span parent);
```

Comment:
@michaelsembwever:

> Do we have any examples of this?

There's one written up here: http://opentracing.io/use-cases/#setting-sampling-priority-before-the-trace-starts

I can see what you mean about the builder pattern, esp since that's so widely used in Java-land. The alternative of passing a hodge-podge of `null`s to `startSpan` is a little gross. Then again, what a bummer to add a builder pattern just for the sake of this one esoteric corner-case...

I can't claim to have particularly strong feelings about this one, but hopefully the above is motivating in some way. My understanding is that this is all somewhat related to the `debug=true` case in Zipkin traces... maybe that rings a bell?

> When would the caller want to decide which Span implementation to use? (that seems like an anti-pattern).

No, that's not what I meant... it's the *Tracer* that would decide which Span implementation to use, and it would do so by consulting these tags provided at `startSpan()` time.

Consider the case where the Tracer impl has a zipkin-like sampling scheme. Let's further assume that the Tracer impl has two concrete Span subclasses it uses: `SampledSpan` and `UnsampledSpan`. Now, what if that Tracer impl wants to allow its user to request a sampled trace without exposing anything about `SampledSpan`/`UnsampledSpan`: this is what [the opentracing.io example](http://opentracing.io/use-cases/#setting-sampling-priority-before-the-trace-starts) is about, of course. So the user can write something like...

```
Span s = tracer.startSpan(
    "...",
    null,
    // Not valid Java, just trying to communicate the idea:
    {"sampling.priority": 1});
```

... and the `tracer` will use the tag to select the `SampledSpan` impl.

Hope that makes the motivation clearer.